website/docs/guides/migration/versions/04-upgrading-to-dbt-utils-v1.0.md

```diff
@@ -0,0 +1,140 @@
+- The `insert_by_period` materialization has been moved to the [experimental-features repo](https://github.com/dbt-labs/dbt-labs-experimental-features/tree/main/insert_by_period). To continue to use it, add the below to your packages.yml file:
+
+```yaml
+packages:
+  - git: https://github.com/dbt-labs/dbt-labs-experimental-features
+    subdirectory: insert_by_period
+    revision: XXXX #optional but highly recommended. Provide a full git sha hash, e.g. 1c0bfacc49551b2e67d8579cf8ed459d68546e00. If not provided, uses the current HEAD.
+```
+
+- `safe_add()` only works with a list of arguments; use `{{ dbt_utils.safe_add(['column_1', 'column_2']) }}` instead of varargs `{{ dbt_utils.safe_add('column_1', 'column_2') }}`.
+- Several long-promised deprecations to `deduplicate()` have been applied:
+    - The `group_by` argument is replaced by `partition_by`.
+    - `relation_alias` is removed. If you need an alias, you can pass it directly to the `relation` argument.
+    - `order_by` is now mandatory. Pass a static value like `1` if you don’t care how they are deduplicated.
+- The deprecated `table` argument has been removed from `unpivot()`. Use `relation` instead.
+
+
+## Resolving error messages
+After upgrading, these are common error messages you may encounter, along with their resolutions.
+<details>
+	<summary><code>dict object has no attribute MACRO_NAME</code></summary>
+	<div>
+		<p><b>Cause</b>: No macro called <code>MACRO_NAME</code> exists. This is most likely because the macro has moved to the <code>dbt</code> namespace (see above). It could also be because you haven't run dbt deps or have misspelled a macro's name.</p>
+		<p><b>Resolution</b>: For <a href="/reference/dbt-jinja-functions/cross-database-macros">cross-database macros</a>, change <code>dbt_utils.MACRO_NAME()</code> to <code>dbt.MACRO_NAME()</code>.</p>
+	</div>
+</details>
+<details>
+	<summary><code>macro 'dbt_macro__generate_surrogate_key' takes not more than 1 argument(s)</code> </summary>
+	<div>
+		<p><b>Cause</b>: <code>generate_surrogate_key()</code> requires a single argument containing a list of columns, not a set of varargs.</p>
+		<p><b>Resolution</b>: Change to <code>dbt_utils.generate_surrogate_key(['column_1', 'column_2'])</code> - note the square brackets. </p>
+	</div>
+</details>
+<details>
+	<summary>The <code>dbt_utils.surrogate_key</code> has been replaced by <code>dbt_utils.generate_surrogate_key</code></summary>
+	<div>
+		<p><b>Cause</b>: <code>surrogate_key()</code> has been replaced. </p>
+		<p><b>Resolution</b>:
+			<ol>
+				<li>Decide whether you need to enable backwards compatibility <a href="#breaking-changes">as detailed above</a>.</li>
+				<li>Find and replace <code>dbt_utils.surrogate_key</code> with <code>dbt_utils.generate_surrogate_key</code>.</li>
+			</ol>
+		</p>
+	</div>
+</details>
+<details>
+	<summary>macro <code>dbt_macro__test_expression_is_true</code> takes no keyword argument <code>condition</code></summary>
```

Comment:
```suggestion
	<summary>The macro <code>dbt_macro__test_expression_is_true</code> takes no keyword argument <code>condition</code></summary>
```

Reply:
Same here, but in doing this you have pointed out that there's inconsistent formatting of the different error messages in the migration guide (some are completely code bocks, some are a mix), so I will fix those!